wrapperclient.go

```diff
@@ -0,0 +1,59 @@
+package sfxlambda
+
+import (
+	"context"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/sfxclient"
+	log "github.com/sirupsen/logrus"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+)
+
+var handlerFuncWrapperClient *sfxclient.HTTPSink
+
+const (
+	sfxAuthToken          = "SIGNALFX_AUTH_TOKEN"
+	sfxIngestEndpoint     = "SIGNALFX_INGEST_ENDPOINT"
+	sfxSendTimeoutSeconds = "SIGNALFX_SEND_TIMEOUT_SECONDS"
+)
+
+func init() {
+	handlerFuncWrapperClient = sfxclient.NewHTTPSink()
+	if handlerFuncWrapperClient.AuthToken = os.Getenv(sfxAuthToken); handlerFuncWrapperClient.AuthToken == "" {
+		log.Errorf("No value for environment variable %s", sfxAuthToken)
+	}
+	if os.Getenv(sfxIngestEndpoint) != "" {
+		if ingestURL, err := url.Parse(os.Getenv(sfxIngestEndpoint)); err == nil {
+			if ingestURL, err = ingestURL.Parse("v2/datapoint"); err == nil {
+				handlerFuncWrapperClient.DatapointEndpoint = ingestURL.String()
+			} else {
+				log.Errorf("Error parsing ingest url path v2/datapoint: %+v", err)
+			}
+		} else {
+			log.Errorf("Error parsing url value %s of environment variable %s. %+v", os.Getenv(sfxIngestEndpoint), sfxIngestEndpoint, err)
+		}
+	}
+	if os.Getenv(sfxSendTimeoutSeconds) != "" {
+		if timeout, err := time.ParseDuration(strings.TrimSpace(os.Getenv(sfxSendTimeoutSeconds)) + "s"); err == nil {
+			handlerFuncWrapperClient.Client.Timeout = timeout
+		} else {
+			log.Errorf("Error parsing timeout value %s of environment variable %s. %+v", os.Getenv(sfxSendTimeoutSeconds), sfxSendTimeoutSeconds, err)
+		}
+	}
+}
+
+func sendDatapoints(ctx context.Context, dps []*datapoint.Datapoint) {
+	now := time.Now()
+	for _, dp := range dps {
+		if dp.Timestamp.IsZero() {
+			dp.Timestamp = now
+		}
+	}
+	go func(ctx context.Context, dps []*datapoint.Datapoint) {
```

Comment:
I'm not sure this should be in a goroutine.  If the lambda is finished executing (i.e. the response is returned to the AWS calling component) this logic could get paused right?  Just make the AddDatapoints call synchronous with this function.

Reply:
No, I think HandlerWrapper.Invoke()'s return is what gets sent back as a response and we are calling HandlerWrapper.SendDatapoints() in HandlerWrapper.Invoke()